### PR TITLE
feat(dataarts): add new resource to manage apis

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Add new resource support for DataArts Studio service.

CheckDeleted
（refresh）
API not found:
image
Workspace not found:
image

（delete）
API not found:
image
Workspace not found:
image

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource support.
2. add related acceptance test and document.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dataarts' TESTARGS='-run=TestAccDataServiceApi_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dataarts -v -run=TestAccDataServiceApi_basic -timeout 360m -parallel 4
=== RUN   TestAccDataServiceApi_basic
=== PAUSE TestAccDataServiceApi_basic
=== CONT  TestAccDataServiceApi_basic
--- PASS: TestAccDataServiceApi_basic (311.65s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts  311.715s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_dataservice_api_test.go [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+		var workspaceId, dlmType, resourceId string
+		rs, ok := s.RootModule().Resources[rName]
+		if !ok {
+			return "", fmt.Errorf("resource (%s) not found: %s", rName, rs)
+		}
+
+		workspaceId = rs.Primary.Attributes["workspace_id"]
+		dlmType = rs.Primary.Attributes["dlm_type"]
+		resourceId = rs.Primary.ID
+		if rs.Primary.Attributes["workspace_id"] == "" || rs.Primary.ID == "" {
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_dataservice_api_test.go]:
```suggestion
		if workspaceId == "" || resourceId == "" {
```

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go [hunk 2]
```diff
@@ -0,0 +1,1198 @@
...
+		getPath += fmt.Sprintf("%s&offset=%d", getPath, offset)
+		requestResp, err := client.Request("GET", getPath, &opt)
+		if err != nil {
+			return nil, fmt.Errorf("error querying datatables (%s) under the database (%s): %s", tableId, databaseName, err)
+		}
+		respBody, err := utils.FlattenResponse(requestResp)
+		if err != nil {
+			return nil, err
+		}
+		if utils.PathSearch("length(tables)", respBody, 0).(float64) < 1 {
```
Comment (deer-hang) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go]:
```suggestion
		if utils.PathSearch("length(tables)", respBody, float64(0)).(float64) < 1 {
```

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go [hunk 3]
```diff
@@ -0,0 +1,1198 @@
...
+			break
+		}
+		if tableDetail := utils.PathSearch(fmt.Sprintf("tables[?table_id=='%s']|[0]", tableId), respBody, nil); tableDetail != nil {
+			return tableDetail, nil
+		}
+		offset++
```
Comment (deer-hang) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go]: `offset++`?

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go [hunk 4]
```diff
@@ -0,0 +1,1198 @@
...
+		d.Set("path", utils.PathSearch("path", apiRecord, nil)),
+		d.Set("request_type", utils.PathSearch("request_type", apiRecord, nil)),
+		d.Set("manager", utils.PathSearch("manager", apiRecord, nil)),
+		d.Set("datasource_config", flattenApiDataSourceConfig(client, workspaceId, utils.PathSearch("datasource_config", apiRecord,
+			make(map[string]interface{})).(map[string]interface{}))),
+		d.Set("request_params", flattenApiRequestParams(utils.PathSearch("request_paras", apiRecord, make([]interface{}, 0)).([]interface{}))),
+		d.Set("backend_config", flattenApiBackendConfig(utils.PathSearch("backend_config", apiRecord, make([]interface{}, 0)).([]interface{}))),
+		// Public attributes.
+		d.Set("create_user", utils.PathSearch("create_user", apiRecord, nil)),
+		d.Set("created_at", utils.FormatTimeStampRFC3339(int64(utils.PathSearch("create_time", apiRecord, 0).(float64))/1000, false)),
```
Comment (deer-hang) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go]: Ditto, check other parts.

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go]: done

File: huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_dataservice_api_test.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_dataservice_api_test.go]: sure

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go]: that was a mistake and fixed

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go]: done

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dataarts' TESTARGS='-run=TestAccDataServiceApi_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dataarts -v -run=TestAccDataServiceApi_basic -timeout 360m -parallel 4
=== RUN   TestAccDataServiceApi_basic
=== PAUSE TestAccDataServiceApi_basic
=== CONT  TestAccDataServiceApi_basic
--- PASS: TestAccDataServiceApi_basic (4614.54s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts  4614.593s
```

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dataarts' TESTARGS='-run=TestAccDataServiceApi_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dataarts -v -run=TestAccDataServiceApi_basic -timeout 360m -parallel 4
=== RUN   TestAccDataServiceApi_basic
=== PAUSE TestAccDataServiceApi_basic
=== CONT  TestAccDataServiceApi_basic
--- PASS: TestAccDataServiceApi_basic (4663.55s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dataarts  4663.605s
```

File: huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_dataservice_api_test.go [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/dataarts"
+)
+
+func getDataServiceApiStreamFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := cfg.NewServiceClient("dataarts", acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating EG client: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_dataservice_api_test.go]:
```suggestion
		return nil, fmt.Errorf("error creating DataArts Studio client: %s", err)
```

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go [hunk 6]
```diff
@@ -0,0 +1,1205 @@
...
+		return diag.Errorf("error creating API: %s", err)
+	}
+
+	respBody, err := utils.FlattenResponse(requestResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	apiId, err := jmespath.Search("id", respBody)
+	if err != nil {
```
Comment (deer-hang) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go]:
```suggestion
	if err != nil || apiId == nil {
```

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go [hunk 7]
```diff
@@ -0,0 +1,1205 @@
...
+	opt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+			"workspace":    workspaceId,
+		},
+	}
+	var offset float64
+	for {
+		getPath += fmt.Sprintf("%s&offset=%v", getPath, offset)
```
Comment (deer-hang) [inline huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go]:
```suggestion
		eachPath := fmt.Sprintf("%s&offset=%d", getPath, offset)
```

Comment (Lance52259) [reply huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go]: The number of type float64 can only use `%v` to format

File: huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go (hunk 6 above)
Comment (Lance52259) [reply huaweicloud/services/dataarts/resource_huaweicloud_dataarts_dataservice_api.go]: done.

**Pull request merged**